vllm/config.py

```diff
@@ -152,6 +153,7 @@ def __init__(
                                     code_revision, rope_scaling, rope_theta)
         self.hf_text_config = get_hf_text_config(self.hf_config)
         self.dtype = _get_and_verify_dtype(self.hf_text_config, dtype)
+        self.model_mode = ModelMode.DECODER
```

Comment:
Maybe can remove `_verify_model_mode` and just set it here.

Reply:
done.